Modules/Scripted/SegmentStatistics/SegmentStatisticsCalculators/SegmentStatisticsCalculatorBase.py

```diff
@@ -0,0 +1,201 @@
+import vtk
+import qt
+import slicer
+
+
+class SegmentStatisticsCalculatorBase(object):
+  """Base class for statistics calculators operating on segments.
+  Derived classes should specify: self.name, self.keys, self.defaultKeys
+  and implement: computeStatistics, getMeasurementInfo
+  """
+
+  @staticmethod
+  def initCodedEntry(codeValue, codingScheme, codeMeaning, returnAsString=False):
+    entry = slicer.vtkCodedEntry()
+    entry.SetValueSchemeMeaning(codeValue, codingScheme, codeMeaning)
+    return entry if not returnAsString else entry.GetAsString()
+
+  @staticmethod
+  def getDICOMTriplet(codeValue, codingSchemeDesignator, codeMeaning):
+    """Utility method to form DICOM triplets in getMeasurementInfo"""
+    return {'CodeValue':codeValue,
+            'CodingSchemeDesignator': codingSchemeDesignator,
+            'CodeMeaning': codeMeaning}
+
+  @staticmethod
+  def generateMeasurementInfo(name, description, units, quantityCode, unitsCode, measurementMethodCode=None,
```

Comment:
What type quantityCode and unitsCode is expected in? VTK, string, or DICOM?
At least the argument name should reflect it, but preferably method documentation should specify it.